pkg/server/devfile-handler.go

```diff
@@ -191,7 +191,7 @@ func getDeployResource(data devfileForm, devfileObj parser.DevfileObj, filterOpt
 
 	deployment := generator.GetDeployment(deployParams)
 
-	return *deployment, nil
+	return *deployment, err
```

Comment:
This looks suspicious. Don't we return `err` above if not nil? Do we need to check the error from `GetDeployment`?

Reply:
yes, I caught this in the 4.9 PR too..I need to revert this file to its previous state

Reply:
I downgraded the `devfile library` dependency since it's not needed for this backport.  Previous changes were due to cherry picking from the latest `registry library`  which pulled in a newer `devfile library` that returned an error for `GetDeployment` - older version version does not return an error.   In order to fix this properly, I had to create a branch release of the registry library rather than cherry pick the changes